File: LeetCode_Practice/C++_Solutions/Arrays/_1480_RunningSumof1DArray.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> res;
        int runningSum = 0;

        for (int i=0; i < nums.size(); i++)
        {
            runningSum += nums[i];
            res.push_back(runningSum);
        }

        return res;
    }
};

File: LeetCode_Practice/Java_Solutions/Arrays/_1480_RunningSumof1DArray.java
package LeetCode_Practice.Java_Solutions.Arrays;

import java.util.Scanner;

public class _1480_RunningSumof1DArray {
    static int[] runningSum(int[] nums) {
        int res[] = new int[nums.length];
        int runningSum = 0;

        for (int i=0; i < nums.length; i++) {
            runningSum += nums[i];
            res[i] = runningSum;
        }

        return res;
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Enter size of array : ");
        int n = input.nextInt();
        
        int[] nums = new int[n];
        System.out.print("Enter elements : ");
        for (int i=0; i<n; i++)
            nums[i] = input.nextInt();
        
        nums = runningSum(nums);
        System.out.print("The resulting array is : ");
        for (int i=0; i<n; i++)
            System.out.print(nums[i] + ", ");

        input.close();
    }
}
